README.md

```diff
@@ -1,2 +1,254 @@
+  - [Configuration Options](https://github.com/mayankshah1607/linkerd2-conformance#configuration-options)
+- [Usage](https://github.com/mayankshah1607/linkerd2-conformance#usage)
+  - [Using the Sonobuoy CLI](https://github.com/mayankshah1607/linkerd2-conformance#using-the-sonobuoy-cli)
+  - [Running the tests using Docker]()
+  - [Running the tests locally](https://github.com/mayankshah1607/linkerd2-conformance#running-the-tests-locally)
+- [Adding new tests](https://github.com/mayankshah1607/linkerd2-conformance#adding-new-tests)
+  - [Initial bootstrapping](https://github.com/mayankshah1607/linkerd2-conformance#1-initial-bootstrapping)
+  - [Adding tests](https://github.com/mayankshah1607/linkerd2-conformance#2-adding-tests)
+  - [Wiring up newly added tests](https://github.com/mayankshah1607/linkerd2-conformance#3-wiring-up-newly-added-test)
+
+## Repository Structure
+
+- [`tests`](https://github.com/mayankshah1607/linkerd2-conformance/tree/master/specs) contains the tests for each of the features organized into separate packages
+- [`sonobuoy`](https://github.com/mayankshah1607/linkerd2-conformance/tree/master/sonobuoy) contains the items required to be able to run the tests as a [Sonobuoy](https://github.com/vmware-tanzu/sonobuoy) plugin
+- [`utils`](https://github.com/mayankshah1607/linkerd2-conformance/tree/master/utils) contains helper functions that can be used while writing conformance tests
+- [`bin`](https://github.com/mayankshah1607/linkerd2-conformance/blob/master/bin) contains useful helper scripts to build/push the Docker image and running the tests
+- [`test_template`](https://github.com/mayankshah1607/linkerd2-conformance/tree/master/test_template) contains a sample template for adding new tests under the `tests/` folder
+
+## Configuring your tests
+
+The conformance tests can be easily configured by specifying a `config.yaml` that holds the configuration values. This includes things like Linkerd version, Linkerd add-ons, test specific configuration, binary path, etc.  
+
+Run the command below to pull up a sample configuration file, `config.yaml`, and modify it according to your requirements. The tests shall read this YAML file during runtime and run accordingly.
+
+```bash
+$ curl -sL https://raw.githubusercontent.com/mayankshah1607/linkerd2-conformance/master/config.yaml > config.yaml
+```
+
+### Configuration options
+
+This section describes the various configuration options and its default values. Not providing a configuration file, or providing a partial configuration file will result in the tests running on default settings as described below
+
+| Option | Description | Default value |
+|-|-|-|
+| `linkerdVersion` | The linkerd2 binary version to use | Latest stable release |
+| `linkerdBinaryPath` | If specified, the tests use the binary installed in the directory. It is recommended that this is left unspecified while using Sonobuoy or if upgrade tests are enabled | `$HOME/.linkerd2/bin/linkerd` | 
```

Comment:
The `linkerdBinaryPath` doesn't seem to work for me, when I run the tests locally:

```bash
$ cat config.yaml| grep linkerdBinaryPath                                            
linkerdBinaryPath: /home/isim/.linkerd2/bin/linkerd  # specify default location in config.yaml

$ bin/run-all
failed to initialize test helper or config: failed to parse YAML: yaml: unmarshal errors:
  line 2: field linkerdBinaryPath not found in type utils.ConformanceTestOptionsFAIL    github.com/linkerd/linkerd2-conformance/tests/lifecycle 0.014s
FAIL
failed to initialize test helper or config: failed to parse YAML: yaml: unmarshal errors:
  line 2: field linkerdBinaryPath not found in type utils.ConformanceTestOptionsFAIL    github.com/linkerd/linkerd2-conformance/tests/inject    0.014s
FAIL
```

Reply:
This is fixed!